version: "3.3"

services:
  psql:
    image: postgres:14.1
    ports: 
      - "5479:5432"
    environment:
      - POSTGRES_DB=test_db
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres

  django:
    build:
      context: .
      dockerfile: docker/local.Dockerfile
    command: python manage.py runserver 0.0.0.0:8000
    env_file: 
      - ./.env.dev
    volumes:
      - .:/app
    ports:
      - "8000:8000"
    depends_on:
      - psql
      - redis
    restart: on-failure

  redis:
    image: redis:alpine
  celery:
    build:
      context: .
      dockerfile: docker/local.Dockerfile
    command: celery -A app.core worker -l info
    volumes:
      - .:/app
    env_file: 
      - ./.env.dev
    depends_on:
      - redis
  celery-beat:
    build:
      context: .
      dockerfile: docker/local.Dockerfile
    command: celery -A app.core beat -l info
    volumes:
      - .:/app
    env_file: 
      - ./.env.dev
    depends_on:
      - redis